src/Nethermind/Nethermind.Core.Test/Json/ByteArray32ConverterTests.cs

```diff
@@ -12,13 +12,25 @@ public class Bytes32ConverterTests : ConverterTestBase<byte[]>
     {
         [TestCase(null)]
         [TestCase(new byte[0])]
-        [TestCase(new byte[] { 1 })]
         public void Empty_value(byte[] values)
         {
             TestConverter(
                 values,
                 (before, after) => Bytes.AreEqual(before.WithoutLeadingZeros(), after.WithoutLeadingZeros()),
                 new Bytes32Converter());
         }
+
+        [TestCase(new byte[] { 1 })]
+        [TestCase(new byte[] { 0, 1 })]
+        [TestCase(new byte[] { 0, 0, 1 })]
+        [TestCase(new byte[] { 0, 0, 1, 0 })]
+        [TestCase(new byte[] { 0, 0, 1, 0, 0 })]
+        public void NonEmpty_value(byte[] values)
+        {
+            TestConverter(
+                values,
+                (before, after) => Bytes.AreEqual(before.WithoutLeadingZeros(), after.WithoutLeadingZeros()),
+                new Bytes32Converter());
+        }
```

Comment:
Is it the same test? Should we combine them, change name, name testcases?

Reply:
Combined